### PR TITLE
Restrict validity of browser certs to 2 years

### PR DESCRIPTION
**Pull request by timuthy**
**What this PR does / why we need it**:
This PR makes Gardener to restrict the validity of certificates for the following service to 2 years:
- Garfana (operator / user)
- Prometheus
- Kibana

This is especially required since Apple's announcement:
https://support.apple.com/en-us/HT210176

**Special notes for your reviewer**:
Follow up item: #1684

/cc @dguendisch 

**Release note**:

```improvement operator
Server certificates for Grafana (operator / user), Prometheus and Kibana endpoints of Shoots are now created with a validity of 2 years (unlike 10 years previously). 
```

```improvement operator
A bug has been fixed which caused the Grafana ingresses of shoots to serve the Kubernetes Fake Certificate instead of a certificate signed by the cluster CA. 
```

### CODE REVIEW

**Review by vpnachev — DISMISSED** [review]

**Comment by rfranzke** [comment]
Should we have some automated deletion of the secrets so that they get recreated for existing clusters as well?

**Comment by vpnachev** [comment]
Theoretically there are clusters that will be affected by this issue in the next almost 10 years. But we will have to recreate the certificates one more time when we start signing them with public CA(#1482 ), so we can postpone it.

**Comment by timuthy** [comment]
> Should we have some automated deletion of the secrets so that they get recreated for existing clusters as well?

Do you mean like a one time deletion logic which will be removed after a few released and is specific to that case, e.g. `if creationTimestamp > 1. Jul 2019 && vaild => 2029`? Or rather a general cert rotation logic (triggered by annotation), so it'd be up to the Gardener operator to add it to the affected shoots?

**Comment by rfranzke** [comment]
No, the one time action. AFAIK there are shoots whose Grafana/Kibana Dashboards cannot be opened atm, right?

**Comment by timuthy** [comment]
It depends on your OS and Browser which is why I considered it a manual change by the Gardener operator, i.e. delete the secret and trigger reconciliation. But we can also build in the one time action :)

**Comment by rfranzke** [comment]
Hm, I think we can simply delete the existing TLS certificate secrets and let Gardener recreate them with the new validity date. No manual action, little effort on our side, huge effects. WDYT?

**Comment by timuthy** [comment]
OK will do

**Comment by timuthy** [comment]
@rfranzke I addressed your comment. Can you please have another look?

File: pkg/utils/secrets/generate.go [hunk 1]
```diff
@@ -67,10 +78,22 @@ func GenerateClusterSecrets(ctx context.Context, k8sClusterClient kubernetes.Int
 					Name:      s.GetName(),
 					Namespace: namespace,
 				},
-				Type: secretType,
-				Data: obj.SecretData(),
 			}
-			err = k8sClusterClient.Client().Create(ctx, secret)
+
+			err = kutil.CreateOrUpdate(ctx, k8sClusterClient.Client(), secret, func() error {
```
Comment (rfranzke) [inline pkg/utils/secrets/generate.go]: Can we move the adding of the label a separate step and to `pkg/botanist/secrets.go`? This location here is more meant to be a utility library to generate secrets - it should not contain any specifics.

**Review by rfranzke — CHANGES_REQUESTED** [review]

File: pkg/utils/secrets/generate.go (hunk 1 above)
Comment (timuthy) [reply pkg/utils/secrets/generate.go]: Yes, sure - thanks for pointing that out. I've changed it.

**Review by rfranzke — APPROVED** [review]

**Pull request merged**